specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-05-01-preview/FileImports.json

```diff
@@ -0,0 +1,560 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataFilter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataOrderBy"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataTop"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataSkipToken"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/FileImportList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/fileImports/{fileImportId}": {
+      "get": {
+        "x-ms-examples": {
+          "Get a file import.": {
+            "$ref": "./examples/fileImports/GetFileImportById.json"
+          }
+        },
+        "tags": [
+          "FileImports"
+        ],
+        "description": "Gets a file import.",
+        "operationId": "FileImport_Get",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/FileImportId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/FileImport"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
```

Comment:
>put [](http://example.com/codeflow?start=7&length=3)

Having a PUT API without an input body implies that Export Template would not export your resource and customers would not be able to use Export Template functionality.

If you are sure that you don't want to include a PUT body, one workaround is to actually create a body but mark the properties as readonly: true -> one of the properties could be "provisioningState" this way your API will have a PUT body but since the property is readonly it won't require customers to pass "provisioningState" for example.

One example you can follow is here: 

[ApplicationSecurityGroups](https://github.com/Azure/azure-rest-api-specs/blob/88e7838a09868a51de3894114355c75929847a46/specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/applicationSecurityGroup.json#L129) and look at its PUT [body](https://github.com/Azure/azure-rest-api-specs/blob/88e7838a09868a51de3894114355c75929847a46/specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/applicationSecurityGroup.json#L318)

Reply:
Hi,
our PUT call does have a body. the body contains the "fileImport" object (as specified in "parameters").
I'm not sure if you mean we need to specifically add a "body" field, but we just followed what all of our other APIs do. if you look at any other PUT call in our RP, all of them mention the body in the "parameters" section.